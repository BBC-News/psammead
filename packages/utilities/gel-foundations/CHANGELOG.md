# Gel Foundations Changelog

| Version | Description |
|---------|-------------|
| 0.1.4   | [PR#224](https://github.com/BBC-News/psammead/pull/224) Add tests for the exported values, coverage 100% :tada: |
| 0.1.1   | [PR#212](https://github.com/BBC-News/psammead/pull/212) Update package description and README. |
| 0.1.0   | [PR#221](https://github.com/BBC-News/psammead/pull/221) Create initial package, pulled in from gel-constants and gel-foundations-styled-components packages. |

