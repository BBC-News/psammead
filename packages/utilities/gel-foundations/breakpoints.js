module.exports = require('./dist/breakpoints'); // eslint-disable-line import/no-unresolved
