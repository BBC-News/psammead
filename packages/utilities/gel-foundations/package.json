{
  "name": "drew-testing-123",
  "version": "1.0.6",
  "main": "dist/index.cjs.js",
  "module": "dist/index.es.js",
  "description": "A range of string constants for use in CSS, intended to help implement BBC GEL-compliant webpages and components.",
  "repository": {
    "type": "git",
    "url": "https://github.com/bbc/psammead/tree/latest/packages/utilities/gel-foundations"
  },
  "author": {
    "name": "Psammead Maintainers",
    "email": "PsammeadMaintainers@bbc.co.uk"
  },
  "files": [
    "dist/"
  ],
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/bbc/psammead/issues"
  },
  "homepage": "https://github.com/bbc/psammead/blob/latest/packages/utilities/gel-foundations/README.md",
  "keywords": [
    "bbc",
    "utilities",
    "gel",
    "breakpoints",
    "grid",
    "spacing",
    "typography"
  ],
  "devDependencies": {
    "@bbc/psammead-test-helpers": "^1.0.1"
  }
}
