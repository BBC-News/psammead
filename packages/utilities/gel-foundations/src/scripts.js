export latin from './scripts/latin-and-cyrillic';
export cyrillic from './scripts/latin-and-cyrillic';
export latinDiacritics from './scripts/latin-with-diacritics';
export arabic from './scripts/arabic';
export arabicPashto from './scripts/arabic-pashto';
export bengali from './scripts/bengali';
export burmese from './scripts/burmese';
export hindi from './scripts/devanagari-and-gurmukhi';
export nepali from './scripts/devanagari-and-gurmukhi';
export sinhalese from './scripts/sinhalese';
export tamil from './scripts/tamil';
export thai from './scripts/thai';
export chinese from './scripts/no-ascenders-or-descenders';
export korean from './scripts/no-ascenders-or-descenders';
export ethiopic from './scripts/no-ascenders-or-descenders';
