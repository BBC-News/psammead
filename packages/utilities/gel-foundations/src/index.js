import * as breakpoints from './breakpoints';
import * as propTypes from './prop-types';
import * as scripts from './scripts';
import * as spacings from './spacings';
import * as typography from './typography';

export { breakpoints };
export { propTypes };
export { scripts };
export { spacings };
export { typography };
