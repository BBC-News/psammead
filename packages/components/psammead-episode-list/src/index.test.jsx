import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { render } from '@testing-library/react';
import * as scripts from '@bbc/gel-foundations/scripts';
import '@testing-library/jest-dom/extend-expect';

import {
  renderEpisodes,
  renderVideoEpisodes,
  exampleEpisodes,
  exampleVideoEpisodes,
} from './fixtures';
import EpisodeList from '.';

describe('Episode List ', () => {
  shouldMatchSnapshot(
    'should render video episodes correctly',
    renderEpisodes({
      episodes: exampleVideoEpisodes,
      script: scripts.latin,
      service: 'news',
      dir: 'ltr',
    }),
  );

  shouldMatchSnapshot(
    'should render radio episodes correctly',
    renderEpisodes({
      episodes: exampleEpisodes,
      script: scripts.latin,
      service: 'news',
      dir: 'ltr',
    }),
  );

  it('should render the list', () => {
    const { getByRole } = render(
      renderEpisodes({
        episodes: exampleEpisodes,
        script: scripts.latin,
        service: 'news',
        dir: 'ltr',
      }),
    );

    expect(getByRole('list')).toBeInTheDocument();
  });

  it('should render the list item', () => {
    const { getAllByRole } = render(
      renderEpisodes({
        episodes: exampleEpisodes,
        script: scripts.latin,
        service: 'news',
        dir: 'ltr',
      }),
    );

    expect(getAllByRole('listitem')[0]).toBeInTheDocument();
  });

  it('should render the brand title', () => {
    const { getByText } = render(
      renderEpisodes({
        episodes: exampleEpisodes,
        script: scripts.latin,
        service: 'news',
        dir: 'ltr',
      }),
    );

    expect(getByText('Le Journal')).toBeInTheDocument();
  });

  it('should render the episode title', () => {
    const { getByText } = render(
      renderEpisodes({
        episodes: exampleEpisodes,
        script: scripts.latin,
        service: 'news',
        dir: 'ltr',
      }),
    );

    expect(
      getByText("Le premier rendez-vous d'information de la soirée."),
    ).toBeInTheDocument();
  });

  it('should render the link', () => {
    const { getByText } = render(
      renderEpisodes({
        episodes: exampleEpisodes,
        script: scripts.latin,
        service: 'news',
        dir: 'ltr',
      }),
    );

    expect(
      getByText(exampleEpisodes[0].brandTitle).closest('a'),
    ).toHaveAttribute('href', exampleEpisodes[0].url);
  });

  it('should render the media indicator', () => {
    const { container } = render(
      renderEpisodes({
        episodes: exampleEpisodes,
        script: scripts.latin,
        service: 'news',
        dir: 'ltr',
      }),
    );
    const svgs = container.querySelectorAll('svg');

    expect(svgs).toHaveLength(3);
  });

  it('should render the duration', () => {
    const { getAllByText } = render(
      renderEpisodes({
        episodes: exampleEpisodes,
        script: scripts.latin,
        service: 'news',
        dir: 'ltr',
      }),
    );

    expect(getAllByText('Durée 59:00')[0]).toBeInTheDocument();
  });

  it('should render the date', () => {
    const { getAllByText } = render(
      renderEpisodes({
        episodes: exampleEpisodes,
        script: scripts.latin,
        service: 'news',
        dir: 'ltr',
      }),
    );

    expect(getAllByText('4 Avril 2020, 14:00')[0]);
  });

  it('should render the correct number of episodes', () => {
    const { getAllByRole } = render(
      renderEpisodes({
        episodes: exampleEpisodes,
        script: scripts.latin,
        service: 'news',
        dir: 'ltr',
      }),
    );

    expect(getAllByRole('listitem').length).toEqual(3);
  });

  it('should not render a list when there is only one episode', () => {
    const { queryByRole } = render(
      renderEpisodes({
        episodes: [exampleEpisodes[0]],
        script: scripts.latin,
        service: 'news',
        dir: 'ltr',
      }),
    );

    expect(queryByRole('list')).not.toBeInTheDocument();
    expect(queryByRole('listitem')).not.toBeInTheDocument();
  });

  it('should correctly handle images', () => {
    const { container } = render(
      renderVideoEpisodes({
        episodes: [exampleVideoEpisodes[0]],
        script: scripts.latin,
        service: 'news',
        dir: 'ltr',
      }),
    );

    expect(
      container.querySelector(`img[src='${exampleVideoEpisodes[0].image}']`),
    ).toBeInTheDocument();
    expect(
      container.querySelector(`img[alt='${exampleVideoEpisodes[0].altText}']`),
    ).toBeInTheDocument();
  });

  it('should include the data-e2e attribute if passed', () => {
    const { container } = render(
      renderEpisodes({
        episodes: exampleEpisodes,
        script: scripts.latin,
        service: 'news',
        dir: 'ltr',
        ulProps: { 'data-e2e': 'recent-episode-list' },
        liProps: { 'data-e2e': 'recent-episode-list-item' },
      }),
    );

    expect(container.querySelector('ul')).toHaveAttribute(
      'data-e2e',
      'recent-episode-list',
    );
    expect(container.querySelector('li')).toHaveAttribute(
      'data-e2e',
      'recent-episode-list-item',
    );
  });

  it('should render a span with role=text to avoid text splitting in screenreaders', () => {
    const { getAllByRole } = render(
      renderEpisodes({
        episodes: exampleEpisodes,
        script: scripts.latin,
        service: 'news',
        dir: 'ltr',
      }),
    );

    expect(getAllByRole('text')[0].closest('a')).toBeInTheDocument();
  });

  it('should render the vertical separator with an aria-hidden attribute', async () => {
    const { container } = render(<EpisodeList.VerticalSeparator />);
    const spanEl = container.getElementsByTagName('span')[0];
    expect(spanEl.hasAttribute('aria-hidden')).toEqual(true);
  });
});
