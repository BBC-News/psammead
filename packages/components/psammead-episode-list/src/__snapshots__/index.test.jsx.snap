// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episode List  should render radio episodes correctly 1`] = `
.emotion-0 {
  list-style: none;
  padding: 0;
  margin: 0;
}

.emotion-2 {
  padding: 1rem 0;
  line-height: 0;
}

.emotion-2:first-child {
  padding-top: 0;
}

.emotion-2:last-child {
  padding-bottom: 0;
}

.emotion-2:not(:last-child) {
  border-bottom: 1px #BABABA solid;
}

.emotion-4 {
  position: relative;
  padding-left: 4rem;
}

.emotion-6 {
  line-height: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-6:before {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  content: '';
  overflow: hidden;
  z-index: 1;
}

.emotion-6 .rounded-play-button__ring,
.emotion-6 .rounded-play-button__triangle {
  color: #000;
}

.emotion-6:focus [class*='--hover'],
.emotion-6:hover [class*='--hover'] {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-6:focus .rounded-play-button__ring,
.emotion-6:hover .rounded-play-button__ring,
.emotion-6:focus .rounded-play-button__inner,
.emotion-6:hover .rounded-play-button__inner {
  fill: currentColor;
  color: #B80000;
}

.emotion-6:focus .rounded-play-button__triangle,
.emotion-6:hover .rounded-play-button__triangle {
  fill: transparent;
}

.emotion-6:visited [class*='--visited'] {
  color: #6E6E73;
}

.emotion-8 {
  position: absolute;
  left: 0.5rem;
  top: 0;
}

.emotion-10 {
  display: inline-block;
  width: 2.5rem;
  height: 2.5rem;
}

.emotion-12 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.emotion-14 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #222222;
  display: inline-block;
  width: 100%;
  font-weight: 700;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .emotion-14 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .emotion-14 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

.emotion-18 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #222222;
  display: inline-block;
  width: 100%;
  margin: 0.25rem 0;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .emotion-18 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .emotion-18 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

.emotion-24 {
  font-size: 0.875rem;
  line-height: 1.125rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #6E6E73;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .emotion-24 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .emotion-24 {
    font-size: 0.8125rem;
    line-height: 1rem;
  }
}

.emotion-50 {
  display: inline;
}

.emotion-52 {
  font-size: 0.875rem;
  line-height: 1.125rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #6E6E73;
  padding-left: 0.5rem;
  margin-left: 0.5rem;
  border-left: 0.0625rem solid #BABABA;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .emotion-52 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .emotion-52 {
    font-size: 0.8125rem;
    line-height: 1rem;
  }
}

<div>
  <ul
    class="emotion-0 emotion-1"
    role="list"
  >
    <li
      class="emotion-2 emotion-3"
    >
      <div
        class="emotion-4 emotion-5"
        dir="ltr"
      >
        <a
          class="emotion-6 emotion-7"
          href="https://www.bbc.com/blahasda"
        >
          <div
            class="emotion-8 emotion-9"
            dir="ltr"
          >
            <div
              aria-hidden="true"
              class="emotion-10 emotion-11"
            >
              
      
              <svg
                class="rounded-play-button"
                focusable="false"
                viewBox="0 0 40 40"
                width="2.5rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                
        
                <path
                  class="rounded-play-button__inner"
                  d="M20,0C8.971,0,0,8.972,0,20c0,11.028,8.972,20,20,20c11.029,0,20.002-8.972,20.002-20C40.002,8.972,31.028,0,20,0z
        M15.799,26.939V13.078l11.264,6.93L15.799,26.939z"
                  fill="transparent"
                />
                
        
                <path
                  class="rounded-play-button__ring"
                  d="M20,40C8.971,40,0,31.028,0,20C0,8.972,8.971,0,20,0c11.029,0,20.002,8.972,20.002,20C40.002,31.028,31.028,40,20,40z
        M20,1.765C9.945,1.765,1.764,9.945,1.764,20S9.945,38.234,20,38.234c10.056,0,18.237-8.18,18.237-18.234S30.056,1.765,20,1.765z"
                  fill="currentColor"
                />
                
        
                <polygon
                  class="rounded-play-button__triangle"
                  fill="currentColor"
                  points="15.799,26.939 27.062,20.008 15.799,13.078 "
                />
                
      
              </svg>
              
      
            </div>
          </div>
          <span
            role="text"
          >
            <span
              class="emotion-12 emotion-13"
            >
              Audio, 
            </span>
            <span
              class="episode-list__title--hover episode-list__title--visited emotion-14 emotion-15"
              dir="ltr"
            >
              Magazine de la Culture
            </span>
            <span
              class="emotion-12 emotion-13"
            >
              , 
            </span>
            <span
              class="episode-list__description--hover episode-list__description--visited emotion-18 emotion-19"
              dir="ltr"
            >
              4 Avril 2020, 14:00
            </span>
            <span
              class="emotion-12 emotion-13"
            >
              , 
            </span>
            <span
              class="emotion-12 emotion-13"
            >
               Durée 03,00 
            </span>
            <span
              class="emotion-24 emotion-25"
              dir="ltr"
            >
              <span
                aria-hidden="true"
              >
                 Durée 03:00
              </span>
            </span>
          </span>
        </a>
      </div>
    </li>
    <li
      class="emotion-2 emotion-3"
    >
      <div
        class="emotion-4 emotion-5"
        dir="ltr"
      >
        <a
          class="emotion-6 emotion-7"
          href="https://www.bbc.com/1"
        >
          <div
            class="emotion-8 emotion-9"
            dir="ltr"
          >
            <div
              aria-hidden="true"
              class="emotion-10 emotion-11"
            >
              
      
              <svg
                class="rounded-play-button"
                focusable="false"
                viewBox="0 0 40 40"
                width="2.5rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                
        
                <path
                  class="rounded-play-button__inner"
                  d="M20,0C8.971,0,0,8.972,0,20c0,11.028,8.972,20,20,20c11.029,0,20.002-8.972,20.002-20C40.002,8.972,31.028,0,20,0z
        M15.799,26.939V13.078l11.264,6.93L15.799,26.939z"
                  fill="transparent"
                />
                
        
                <path
                  class="rounded-play-button__ring"
                  d="M20,40C8.971,40,0,31.028,0,20C0,8.972,8.971,0,20,0c11.029,0,20.002,8.972,20.002,20C40.002,31.028,31.028,40,20,40z
        M20,1.765C9.945,1.765,1.764,9.945,1.764,20S9.945,38.234,20,38.234c10.056,0,18.237-8.18,18.237-18.234S30.056,1.765,20,1.765z"
                  fill="currentColor"
                />
                
        
                <polygon
                  class="rounded-play-button__triangle"
                  fill="currentColor"
                  points="15.799,26.939 27.062,20.008 15.799,13.078 "
                />
                
      
              </svg>
              
      
            </div>
          </div>
          <span
            role="text"
          >
            <span
              class="emotion-12 emotion-13"
            >
              Audio, 
            </span>
            <span
              class="episode-list__title--hover episode-list__title--visited emotion-14 emotion-15"
              dir="ltr"
            >
              Le Journal
            </span>
            <span
              class="emotion-12 emotion-13"
            >
              , 
            </span>
            <span
              class="episode-list__description--hover episode-list__description--visited emotion-18 emotion-19"
              dir="ltr"
            >
              Le premier rendez-vous d'information de la soirée.
            </span>
            <span
              class="emotion-12 emotion-13"
            >
              , 
            </span>
            <span
              class="emotion-12 emotion-13"
            >
               Durée 1,30,00 
            </span>
            <span
              class="emotion-24 emotion-25"
              dir="ltr"
            >
              <span
                aria-hidden="true"
              >
                 Durée 1:30:00
              </span>
            </span>
          </span>
        </a>
        <div
          class="emotion-50 emotion-51"
        >
          <time
            class="emotion-52 emotion-25"
            datetime="2020-10-20"
            dir="ltr"
          >
            20 octobre 2020
          </time>
        </div>
      </div>
    </li>
    <li
      class="emotion-2 emotion-3"
    >
      <div
        class="emotion-4 emotion-5"
        dir="ltr"
      >
        <a
          class="emotion-6 emotion-7"
          href="https://www.bbc.com/2"
        >
          <div
            class="emotion-8 emotion-9"
            dir="ltr"
          >
            <div
              aria-hidden="true"
              class="emotion-10 emotion-11"
            >
              
      
              <svg
                class="rounded-play-button"
                focusable="false"
                viewBox="0 0 40 40"
                width="2.5rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                
        
                <path
                  class="rounded-play-button__inner"
                  d="M20,0C8.971,0,0,8.972,0,20c0,11.028,8.972,20,20,20c11.029,0,20.002-8.972,20.002-20C40.002,8.972,31.028,0,20,0z
        M15.799,26.939V13.078l11.264,6.93L15.799,26.939z"
                  fill="transparent"
                />
                
        
                <path
                  class="rounded-play-button__ring"
                  d="M20,40C8.971,40,0,31.028,0,20C0,8.972,8.971,0,20,0c11.029,0,20.002,8.972,20.002,20C40.002,31.028,31.028,40,20,40z
        M20,1.765C9.945,1.765,1.764,9.945,1.764,20S9.945,38.234,20,38.234c10.056,0,18.237-8.18,18.237-18.234S30.056,1.765,20,1.765z"
                  fill="currentColor"
                />
                
        
                <polygon
                  class="rounded-play-button__triangle"
                  fill="currentColor"
                  points="15.799,26.939 27.062,20.008 15.799,13.078 "
                />
                
      
              </svg>
              
      
            </div>
          </div>
          <span
            role="text"
          >
            <span
              class="emotion-12 emotion-13"
            >
              Audio, 
            </span>
            <span
              class="episode-list__title--hover episode-list__title--visited emotion-14 emotion-15"
              dir="ltr"
            >
              Afrique Avenir
            </span>
            <span
              class="emotion-12 emotion-13"
            >
              , 
            </span>
            <span
              class="episode-list__description--hover episode-list__description--visited emotion-18 emotion-19"
              dir="ltr"
            >
              Tout savoir sur les jeunes entrepreneurs africains.
            </span>
            <span
              class="emotion-12 emotion-13"
            >
              , 
            </span>
            <span
              class="emotion-12 emotion-13"
            >
               Durée 59,00 
            </span>
            <span
              class="emotion-24 emotion-25"
              dir="ltr"
            >
              <span
                aria-hidden="true"
              >
                 Durée 59:00
              </span>
            </span>
          </span>
        </a>
        <div
          class="emotion-50 emotion-51"
        >
          <time
            class="emotion-52 emotion-25"
            datetime="2020-10-21"
            dir="ltr"
          >
            21 octobre 2020
          </time>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`Episode List  should render video episodes correctly 1`] = `
.emotion-0 {
  list-style: none;
  padding: 0;
  margin: 0;
}

.emotion-2 {
  padding: 1rem 0;
  line-height: 0;
}

.emotion-2:first-child {
  padding-top: 0;
}

.emotion-2:last-child {
  padding-bottom: 0;
}

.emotion-2:not(:last-child) {
  border-bottom: 1px #BABABA solid;
}

.emotion-4 {
  position: relative;
  padding-left: 4rem;
}

.emotion-6 {
  line-height: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-6:before {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  content: '';
  overflow: hidden;
  z-index: 1;
}

.emotion-6 .rounded-play-button__ring,
.emotion-6 .rounded-play-button__triangle {
  color: #000;
}

.emotion-6:focus [class*='--hover'],
.emotion-6:hover [class*='--hover'] {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-6:focus .rounded-play-button__ring,
.emotion-6:hover .rounded-play-button__ring,
.emotion-6:focus .rounded-play-button__inner,
.emotion-6:hover .rounded-play-button__inner {
  fill: currentColor;
  color: #B80000;
}

.emotion-6:focus .rounded-play-button__triangle,
.emotion-6:hover .rounded-play-button__triangle {
  fill: transparent;
}

.emotion-6:visited [class*='--visited'] {
  color: #6E6E73;
}

.emotion-8 {
  position: absolute;
  left: 0.5rem;
  top: 0;
}

.emotion-10 {
  display: inline-block;
  width: 2.5rem;
  height: 2.5rem;
}

.emotion-12 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.emotion-14 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #222222;
  display: inline-block;
  width: 100%;
  font-weight: 700;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .emotion-14 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .emotion-14 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

.emotion-18 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #222222;
  display: inline-block;
  width: 100%;
  margin: 0.25rem 0;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .emotion-18 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .emotion-18 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

.emotion-24 {
  font-size: 0.875rem;
  line-height: 1.125rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #6E6E73;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .emotion-24 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .emotion-24 {
    font-size: 0.8125rem;
    line-height: 1rem;
  }
}

.emotion-50 {
  display: inline;
}

.emotion-52 {
  font-size: 0.875rem;
  line-height: 1.125rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #6E6E73;
  padding-left: 0.5rem;
  margin-left: 0.5rem;
  border-left: 0.0625rem solid #BABABA;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .emotion-52 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .emotion-52 {
    font-size: 0.8125rem;
    line-height: 1rem;
  }
}

<div>
  <ul
    class="emotion-0 emotion-1"
    role="list"
  >
    <li
      class="emotion-2 emotion-3"
    >
      <div
        class="emotion-4 emotion-5"
        dir="ltr"
      >
        <a
          class="emotion-6 emotion-7"
          href="https://www.bbc.com/blahasda"
        >
          <div
            class="emotion-8 emotion-9"
            dir="ltr"
          >
            <div
              aria-hidden="true"
              class="emotion-10 emotion-11"
            >
              
      
              <svg
                class="rounded-play-button"
                focusable="false"
                viewBox="0 0 40 40"
                width="2.5rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                
        
                <path
                  class="rounded-play-button__inner"
                  d="M20,0C8.971,0,0,8.972,0,20c0,11.028,8.972,20,20,20c11.029,0,20.002-8.972,20.002-20C40.002,8.972,31.028,0,20,0z
        M15.799,26.939V13.078l11.264,6.93L15.799,26.939z"
                  fill="transparent"
                />
                
        
                <path
                  class="rounded-play-button__ring"
                  d="M20,40C8.971,40,0,31.028,0,20C0,8.972,8.971,0,20,0c11.029,0,20.002,8.972,20.002,20C40.002,31.028,31.028,40,20,40z
        M20,1.765C9.945,1.765,1.764,9.945,1.764,20S9.945,38.234,20,38.234c10.056,0,18.237-8.18,18.237-18.234S30.056,1.765,20,1.765z"
                  fill="currentColor"
                />
                
        
                <polygon
                  class="rounded-play-button__triangle"
                  fill="currentColor"
                  points="15.799,26.939 27.062,20.008 15.799,13.078 "
                />
                
      
              </svg>
              
      
            </div>
          </div>
          <span
            role="text"
          >
            <span
              class="emotion-12 emotion-13"
            >
              Audio, 
            </span>
            <span
              class="episode-list__title--hover episode-list__title--visited emotion-14 emotion-15"
              dir="ltr"
            >
              Magazine de la Culture
            </span>
            <span
              class="emotion-12 emotion-13"
            >
              , 
            </span>
            <span
              class="episode-list__description--hover episode-list__description--visited emotion-18 emotion-19"
              dir="ltr"
            >
              4 Avril 2020, 14:00
            </span>
            <span
              class="emotion-12 emotion-13"
            >
              , 
            </span>
            <span
              class="emotion-12 emotion-13"
            >
               Durée 03,00 
            </span>
            <span
              class="emotion-24 emotion-25"
              dir="ltr"
            >
              <span
                aria-hidden="true"
              >
                 Durée 03:00
              </span>
            </span>
          </span>
        </a>
      </div>
    </li>
    <li
      class="emotion-2 emotion-3"
    >
      <div
        class="emotion-4 emotion-5"
        dir="ltr"
      >
        <a
          class="emotion-6 emotion-7"
          href="https://www.bbc.com/1"
        >
          <div
            class="emotion-8 emotion-9"
            dir="ltr"
          >
            <div
              aria-hidden="true"
              class="emotion-10 emotion-11"
            >
              
      
              <svg
                class="rounded-play-button"
                focusable="false"
                viewBox="0 0 40 40"
                width="2.5rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                
        
                <path
                  class="rounded-play-button__inner"
                  d="M20,0C8.971,0,0,8.972,0,20c0,11.028,8.972,20,20,20c11.029,0,20.002-8.972,20.002-20C40.002,8.972,31.028,0,20,0z
        M15.799,26.939V13.078l11.264,6.93L15.799,26.939z"
                  fill="transparent"
                />
                
        
                <path
                  class="rounded-play-button__ring"
                  d="M20,40C8.971,40,0,31.028,0,20C0,8.972,8.971,0,20,0c11.029,0,20.002,8.972,20.002,20C40.002,31.028,31.028,40,20,40z
        M20,1.765C9.945,1.765,1.764,9.945,1.764,20S9.945,38.234,20,38.234c10.056,0,18.237-8.18,18.237-18.234S30.056,1.765,20,1.765z"
                  fill="currentColor"
                />
                
        
                <polygon
                  class="rounded-play-button__triangle"
                  fill="currentColor"
                  points="15.799,26.939 27.062,20.008 15.799,13.078 "
                />
                
      
              </svg>
              
      
            </div>
          </div>
          <span
            role="text"
          >
            <span
              class="emotion-12 emotion-13"
            >
              Audio, 
            </span>
            <span
              class="episode-list__title--hover episode-list__title--visited emotion-14 emotion-15"
              dir="ltr"
            >
              Le Journal
            </span>
            <span
              class="emotion-12 emotion-13"
            >
              , 
            </span>
            <span
              class="episode-list__description--hover episode-list__description--visited emotion-18 emotion-19"
              dir="ltr"
            >
              Le premier rendez-vous d'information de la soirée.
            </span>
            <span
              class="emotion-12 emotion-13"
            >
              , 
            </span>
            <span
              class="emotion-12 emotion-13"
            >
               Durée 1,30,00 
            </span>
            <span
              class="emotion-24 emotion-25"
              dir="ltr"
            >
              <span
                aria-hidden="true"
              >
                 Durée 1:30:00
              </span>
            </span>
          </span>
        </a>
        <div
          class="emotion-50 emotion-51"
        >
          <time
            class="emotion-52 emotion-25"
            datetime="2020-10-20"
            dir="ltr"
          >
            20 octobre 2020
          </time>
        </div>
      </div>
    </li>
    <li
      class="emotion-2 emotion-3"
    >
      <div
        class="emotion-4 emotion-5"
        dir="ltr"
      >
        <a
          class="emotion-6 emotion-7"
          href="https://www.bbc.com/2"
        >
          <div
            class="emotion-8 emotion-9"
            dir="ltr"
          >
            <div
              aria-hidden="true"
              class="emotion-10 emotion-11"
            >
              
      
              <svg
                class="rounded-play-button"
                focusable="false"
                viewBox="0 0 40 40"
                width="2.5rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                
        
                <path
                  class="rounded-play-button__inner"
                  d="M20,0C8.971,0,0,8.972,0,20c0,11.028,8.972,20,20,20c11.029,0,20.002-8.972,20.002-20C40.002,8.972,31.028,0,20,0z
        M15.799,26.939V13.078l11.264,6.93L15.799,26.939z"
                  fill="transparent"
                />
                
        
                <path
                  class="rounded-play-button__ring"
                  d="M20,40C8.971,40,0,31.028,0,20C0,8.972,8.971,0,20,0c11.029,0,20.002,8.972,20.002,20C40.002,31.028,31.028,40,20,40z
        M20,1.765C9.945,1.765,1.764,9.945,1.764,20S9.945,38.234,20,38.234c10.056,0,18.237-8.18,18.237-18.234S30.056,1.765,20,1.765z"
                  fill="currentColor"
                />
                
        
                <polygon
                  class="rounded-play-button__triangle"
                  fill="currentColor"
                  points="15.799,26.939 27.062,20.008 15.799,13.078 "
                />
                
      
              </svg>
              
      
            </div>
          </div>
          <span
            role="text"
          >
            <span
              class="emotion-12 emotion-13"
            >
              Audio, 
            </span>
            <span
              class="episode-list__title--hover episode-list__title--visited emotion-14 emotion-15"
              dir="ltr"
            >
              Afrique Avenir
            </span>
            <span
              class="emotion-12 emotion-13"
            >
              , 
            </span>
            <span
              class="episode-list__description--hover episode-list__description--visited emotion-18 emotion-19"
              dir="ltr"
            >
              Tout savoir sur les jeunes entrepreneurs africains.
            </span>
            <span
              class="emotion-12 emotion-13"
            >
              , 
            </span>
            <span
              class="emotion-12 emotion-13"
            >
               Durée 59,00 
            </span>
            <span
              class="emotion-24 emotion-25"
              dir="ltr"
            >
              <span
                aria-hidden="true"
              >
                 Durée 59:00
              </span>
            </span>
          </span>
        </a>
        <div
          class="emotion-50 emotion-51"
        >
          <time
            class="emotion-52 emotion-25"
            datetime="2020-10-21"
            dir="ltr"
          >
            21 octobre 2020
          </time>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
